net/network.go

```diff
@@ -116,18 +125,20 @@ func (network *Network) Connect(ctx context.Context, addrs []string) ([]peer.ID,
 		return nil, err
 	}
 
-	output := make([]peer.ID, len(pis))
-	for i, pi := range pis {
-		swrm.Backoff().Clear(pi.ID)
-
-		output[i] = pi.ID
-
-		if err := network.host.Connect(ctx, pi); err != nil {
-			return nil, errors.Wrapf(err, "peer: %s", output[i].Pretty())
+	go func() {
+		for _, pi := range pis {
+			swrm.Backoff().Clear(pi.ID)
+			err := network.host.Connect(ctx, pi)
+			out <- ConnectionResult{
```

Comment:
this is going to block until the receiver receives the value, might be better to send the result to the channel on a goroutine